tests/e2e/default/analytics/analytics.wdio-spec.js

```diff
@@ -105,4 +105,18 @@ describe('Targets', () => {
       'Targets are disabled for admin users. If you need to see targets, login as a normal user.'
     );
   });
+  
+  it('Should show error message for bad config', async () => {
```

Comment:
Hey @jkuester this test fails when i run the entire file but passes if i run it only
I suspect that settings compilation and update is messing with it anything you can see here? Thanks

Reply:
Okay this was really tricky.  After a few hours of tinkering with this, I figured out the source of the problem is in the `tests/utils/cht-conf.js` file!  Basically line 74 is: 

```js
  const appSettings = require(path.join(dir, 'app_settings.json'));
```

For some reason, if this function gets re-run multiple times _in the same test suite_ then `appSettings` does not actually get populated with the _latest_ from the `app_settings.json` file.  Instead, the initial data from the file is getting cached somehow and it will get reused for the subsequent call to this function.  This causes the behavior you describe because the `should display correct message when no target found` test above will call that function and the `appSettings` data gets initialized with the data from that test.  Then, when this test runs, instead of loading the proper config data for this test, the `appSettings` data from the initial test ends up getting used.

To fix it, I think we can just update that line in the `cht-conf.js` to be:

```js
  const appSettings = JSON.parse(fs.readFileSync(path.join(dir, 'app_settings.json'), 'utf-8'));
```